### PR TITLE
support searching Github code, issues or users, not just repositories (which remains the default)

### PR DESCRIPTION
**Pull request by mig5**
As per the request at https://github.com/Netflix/Scumblr/pull/40#issuecomment-155142446, this modifies the Github search provider to be capable of searching for 'users', 'issues', or 'code', not just 'repositories' (the default).

Note: I think this would *look* better as a combobox, but it seems Scumblr's hardcoded any Search Provider's options as a text field. Too much work for me to change this right now, so that's why the 'name' of the option is a bit longer than usual (to explain to the user what the valid options are).

Finally (can't think of a better place to say this), Github's 'code' search requires you to also specify a user, organization, or repository within which to do the code search. The Scumblr user will need to understand that the 'Query' has to be something like 'color+repo:jquery/jquery' as opposed to just 'color', in order to actually get any results.

In other words, it depends on some existing knowledge of the Github search API as defined at https://developer.github.com/v3/search/#search-code

### CODE REVIEW

**Comment by ahoernecke** [comment]
Thanks @mig5
